fiatconnect-api.md

```diff
@@ -391,105 +407,204 @@ The FiatConnect API specifies *two* types of authentication; the first authentic
+###### 3.2.1.5.3.1. Success
 
-Servers MUST recognize the following *registered claims* within the JWT payload.
+On success, the server MUST respond with a `200` status code. The server MUST only respond successfully after performing the following checks:
 
-* `exp`, or *expiration time*
-* `sub`, or *subject*
-* `iss`, or *issuer*
+* A server MUST verify that the signature of the request body is valid and corresponds to the `address` field included in the signed message. If the signature does not
+  correspond to the address field, the server MUST assume that the value given for `address` corresponds to a contract-owned account, and attempt to verify that the
+  signature is valid according to the [EIP-1271](https://eips.ethereum.org/EIPS/eip-1271) standard. If neither of these checks pass, the server MUST reject the request.
 
-Servers MUST also recognize the following *private claims* within the JWT payload.
+* A server MUST verify that the `nonce` field does not correspond to a nonce that has already been used to authenticate an ongoing, valid session.
 
-* `client`, or *client API key*
+* A server MUST verify that the `domain` and `uri` fields correspond exactly to the values expected for the server's hostname/origin.
 
-The semantics of these claims are below:
+* A server MUST verify that the `Version` and `Chain ID` lines in the signed message are exactly `1` and `42220`, respectively.
 
-##### 3.2.1.2.1. `"sub"` Claim
+* If the `issued-at` field in the signed message is after the server's current timestamp, or if the `issued-at` field is after the `expiration-time` field, the server
+  MUST fail the request. If the server's current timestamp is past the `expiration-time` field, the server MUST fail the request. If the `expiration-time` field is
+  greater than four hours (14400 seconds) ahead of the `issued-at` field, the server MUST fail the request.
 
-The `sub`, or *subject* claim represents the user's Celo network address. This is a required claim. If it is missing, the server MUST respond
-to the client with an HTTP `400` error. This claim SHOULD be interpreted and used within server endpoints as a unique user identifier, and as the
-source of/destination for crypto funds during transfers. The address should be formatted as 42 hexadecimal characters prefixed by the string literal
-`0x`. If the `sub` claim does not match this format, the server MUST respond to the client with an HTTP `400` error.
+If all fields are acceptable, and the message is formatted correctly, the server MUST authenticate the user's session, and return an HTTP `200`.
+The server MUST attach the fields from the signed message to the session internally, in order to access data about the user and their session in other endpoints.
 
-##### 3.2.1.2.2. `"iss"` Claim
+Once a server authenticates a session using a particular nonce, the server MUST NOT allow any other sessions to be created using that nonce, until `expiration-time` of the session created
```

Comment:
```suggestion
Once a server authenticates a session using a particular nonce, the server MUST NOT allow any other sessions to be created using the same nonce and address, until `expiration-time` of the session created
```
Servers may segment nonces by client without loosing security. Obviously blocking repeated nonce globally trivially achieves this as well, so maybe not worth pointing out. The one reason this could matter is if clients use timestamp based nonces.